float64 temp_0
float64 temp_1
float64 temp_2 